src/foam/nanos/notification/Notification.js

```diff
@@ -114,5 +120,52 @@ foam.CLASS({
       name: 'slackMessage',
       documentation: 'Message to be sent to Slack if sendSlackMessage is enabled.'
     }
+  ],
+
+  methods: [
+    {
+      name: 'checkOwnership',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      type: 'Boolean',
+      javaCode: `
+        foam.nanos.auth.User user = (foam.nanos.auth.User) x.get("user");
+        if( user == null ) return false;
+
+        Long userId = user.getId();
+        String groupId = user.getGroup();
+
+        return getUserId() != userId && getGroupId() != groupId;
+      `
+    },
+    {
+      name: 'authorizeOnCreate',
+      javaCode: `
+      AuthService auth = (AuthService) x.get("auth");
+      if ( ! auth.check(x, "*") ) throw new AuthorizationException("You don't have permission to create this notification.");
```

Comment:
please create permissions for create/read rather than *

Reply:
added permissions https://github.com/nanoPayinc/NANOPAY/pull/7506